Code/GraphMol/ChemReactions/Reaction.cpp

```diff
@@ -646,4 +646,18 @@ void ChemicalReaction::removeUnmappedProductTemplates(
   res_productTemplates.clear();
 }
 
+void ChemicalReaction::removeAgentTemplates(
+    MOL_SPTR_VECT *targetVector)
+{
+  if(targetVector){
```

Comment:
I think the caller of this function should keep track of what is in the vector or not and it shouldn't be cleared here.